In [ ]:
Simulation
log = []
event = [1,3,6,4] / static = [2, 5]  log = [event[0], static[0], shift]
event = [5,1,2,6] / static = [4, 3]  traces = [5,1]
event = [3,2,4,5] / static = [1, 6]  traces = [4,5]

* 회전 : event = [static[1],다음 값, static[0], 7- 다음 값] / static = [현재 값, 7-현재 값]


event = [1,3,6,4]
static = [2,5]
current = 0
shift = -1
trace = 0
n = 0
while n < R:
    if [event[0], static[0], shift] in log:
        trace += trace*int((R-n+1)/n)
        n += int((R-n+1)/n)*n
        if n == R:
            break
        log = []
    else:
        log.append([event[0], static[0], shift])
    position = 0
    trace += sum(event)*int(C/4)
    C%4번 이동
        current 갱신 
        trace에 현재 값 더하기
        position += shift
    회전/shift = -shift
    n+=1

print(trace)

In [59]:
R, C = map(int, input().split())
event = [1,3,6,4]
static = [2,5]
shift = -1
trace = 0
n = 0
log = []
while n < R:
    if [event[0], static[0], shift] in log:
        trace += trace*int((R-n)/n)
        n += int((R-n)/n)*n
        if n == R:
            break
        log = []
    else:
        log.append([event[0], static[0], shift])
    position = 0
    trace += sum(event)*int(C/4)

    for i in range(0,C%4):
        trace += event[position]
        position += shift
    
    position -= shift

    next_po = (position+1)%4
    current = event[position]
    next_num = event[next_po]
    event = [static[1],next_num, static[0], 7- next_num]
    static = [current, 7-current]
    shift = -shift
    n+=1

print(trace)

737 296
763532


4427

In [55]:
6*122+5

737

int(3/4) 몫 (양수일 때)
3%4   나머지
round  반올림
math.ceil 올림
math.floor 내림

해시 문법
log = [[1,2,3]]
if [1,3,2] in log:
    print('yes')
else:
    print('no')
    

position -= shift               # 반복문에서 마지막 값 고려하기
next_po = (position+1)%4        # 인덱스 값은 리스트 크기 고려
current = event[position]
next_num = event[next_po]       # 변동되는 변수들로 값을 구할 때는 변수를 새로 할당한다!


6 + int(737-6/6)*6 + 737%6          #6 + int(737-6+1/6)*6 + 737%6 (X)  나머지 연산 실수